time/src/test/java/net/jqwik/time/api/DurationTests.java

```diff
@@ -0,0 +1,503 @@
+					@ForAll Random random
+			) {
+
+				Assume.that(start.getNano() > nanosSubtract);
+
+				Duration end = Duration.ofSeconds(start.getSeconds() + 2, start.getNano() - nanosSubtract);
+
+				Arbitrary<Duration> durations = Times.durations().between(start, end);
+
+				assertAllGenerated(durations.generator(1000), random, durationGenerated -> {
+					assertThat(durationGenerated).isNotNull();
+					return true;
+				});
+
+			}
+
+			@Property
+			void worstCaseGeneration9(@ForAll Random random) {
+
+				Duration start = Duration.ofSeconds(0, -101);
+				Duration end = Duration.ofSeconds(0, 100);
+
+				Arbitrary<Duration> durations = Times.durations().between(start, end);
+
+				assertAllGenerated(durations.generator(1000), random, durationGenerated -> {
+					assertThat(durationGenerated).isNotNull();
+					return true;
+				});
+
+			}
+
+			@Provide
+			Arbitrary<Duration> worstCase() {
+				return Times.durations().between(Duration.ZERO, Duration.ofSeconds(Long.MAX_VALUE - 2, 999_999_999));
+			}
+
+			@Provide
+			Arbitrary<Duration> worstCaseNegative() {
+				return Times.durations().between(DefaultDurationArbitrary.DEFAULT_MIN, Duration.ofSeconds(-2, 0));
+			}
+
+		}
+
+	}
+
+	@Group
+	class SimpleAnnotations {
+
+		@Property
+		void validDurationIsGenerated(@ForAll Duration duration) {
+			assertThat(duration).isNotNull();
+		}
+
+	}
+
+	@Group
+	class DurationMethods {
+
+		@Property
+		void between(@ForAll("durations") Duration start, @ForAll("durations") Duration end, @ForAll Random random) {
+
+			Assume.that(start.compareTo(end) <= 0);
+
+			Arbitrary<Duration> durations = Times.durations().between(start, end);
+
+			assertAllGenerated(durations.generator(1000), random, duration -> {
+				assertThat(duration.compareTo(start)).isGreaterThanOrEqualTo(0);
+				assertThat(duration.compareTo(end)).isLessThanOrEqualTo(0);
+			});
+
+		}
+
+		@Property
+		void betweenSame(@ForAll("durations") Duration durationSame, @ForAll Random random) {
+
+			Arbitrary<Duration> durations = Times.durations().between(durationSame, durationSame);
+
+			assertAllGenerated(durations.generator(1000), random, duration -> {
+				assertThat(duration).isEqualTo(durationSame);
+			});
+
+		}
+
+	}
+
+	@Group
+	class Shrinking {
+
+		@Property
+		void defaultShrinking(@ForAll Random random) {
+			DurationArbitrary durations = Times.durations();
+			Duration value = falsifyThenShrink(durations, random);
+			assertThat(value).isEqualTo(Duration.ofSeconds(0, 0));
+		}
+
+		@Property
+		@Disabled("Failing at the moment.")
+		void shrinksToSmallestFailingPositiveValue(@ForAll Random random) {
+			DurationArbitrary durations = Times.durations();
+			TestingFalsifier<Duration> falsifier = duration -> duration.compareTo(Duration.ofSeconds(999_392_192, 709_938_291)) < 0;
+			Duration value = falsifyThenShrink(durations, random, falsifier);
+			assertThat(value).isEqualTo(Duration.ofSeconds(999_392_192, 709_938_291));
+		}
```

Comment:
Same old problem...: Should I accept that shrinking does not work perfectly or what should I do?

```
org.opentest4j.AssertionFailedError:
    expected: 277608H56M32.709938291S
    but was : 277608H56M33S
```

Reply:
I think shrinking to nanos accuracy is not necessary. So it's fine with me.

Reply:
Now I have this error:
```
org.opentest4j.AssertionFailedError:
    expected: 277608H56M33S
    but was : 277608H56M32.709938291S
```

Should I check that the shrunk value is one of this: `277608H56M33S`, `277608H56M32.709938291S` and accept both values?

Reply:
maybe <= *M33

Reply:
Both test cases works now, but `shrinksToSmallestFailingPositiveValue` needs `1m20s` and `shrinksToSmallestFailingNegativeValue` needs `2m51s`.

Should I use only 5-100 tries?

Reply:
I think shrinking test cases are often run with just 100 tries